### PR TITLE
Use server time in startRecord (resolves #507), cleanup

### LINKED ISSUE #507: Accurate time tracking
Please correct me if I'm wrong, but it seems that Kimai alternates between using the server's time and the client machine's time in order to calculate the duration of time tracked. If that is the case, it should be changed so that the start and end time are both drawn from the same source. Assuming that the server and the clients are perfectly synced can lead to errors if, for example, the start time is taken from the server and the end time is taken from the client.

### PR DESCRIPTION
**Pull request by HenryNe**
FIXES #507, #266, #403

Changes proposed in this pull request:
- Use server time in startRecord
- Adjust offset for ticktac (Buzzer)
- "offset" is timediff between server and local, only set ones on page load
- "startsec" is server time
- "now" is local time

Reason for this pull request:
- Fix different time base for start and stop timer

How to test:
1. Force a timediff between server and your local system. For example set your local time 1 hour in future of _same_ timezone. Set local time 14:00 CET, if the server time is 13:00 CET.
2. Press start and stop in same minute ( <60 sec)
3. Failed with old code recorded: In=14:00 Out=13:00 h'm=-1:00 rate=-1,00
4. After this patch recorded: In=13:00 Out=13:00 h'm=0:00 rate=0,00

### CODE REVIEW

File: extensions/ki_timesheets/js/ts_func.js [hunk 1]
```diff
@@ -278,8 +278,7 @@ function ts_ext_recordAgain(project,activity,id) {
     $('#timeSheetEntry'+id+'>td>a.recordAgain>img').attr("src","../skins/"+skin+"/grfx/loading13.gif");
     hour=0;min=0;sec=0;
     now = Math.floor(((new Date()).getTime())/1000);
-    offset = now;
-    startsec = 0;
+    startsec = now - offset;
```
Comment (kevinpapst) [inline extensions/ki_timesheets/js/ts_func.js:281]: Could you please explain this change, I can't wrap my head around it.
What does it have to do with the server time change and what is the outcome of this change?

File: js/main.js [hunk 2]
```diff
@@ -480,8 +479,7 @@ function buzzer_preselect_update_ui(selector,selectedID,updateRecording) {
 // ... so just THX! ;)
 
 function ticktac() {
-    startsecoffset = startsec ? startsec : offset;
-    sek   = Math.floor((new Date()).getTime()/1000)-startsecoffset;
+    sek   = Math.floor((new Date()).getTime()/1000) - startsec - offset;
```
Comment (kevinpapst) [inline js/main.js:482]: Same as above, I get the backend changes, but I am not familiar enough with these old javascript parts. Why did you change these global variables?

**Comment by kevinpapst** [comment]
Ok, this really needs to be fixed. Your solution is simple and effective and I guess we can merge it (after I understood the javascript changes).

We should be consistent and use server time for both actions (start and stop). 
What do you think @simonschaufi ? 

In the future we should still think about passing in a full ISO8601 Date object from javascript to determine the timezone of the user (or use its profile setting) and add that to the record for displaying purpose (but thats a complete different topic).

File: extensions/ki_timesheets/js/ts_func.js (hunk 1 above)
Comment (HenryNe) [inline extensions/ki_timesheets/js/ts_func.js:281]: This change is for the stopwatch near Buzzer.

If you click the record again, then the buzzer will run stopwatch.
The buzzer calculated the timediff from "startsec", but ticktac needs this "startsec" as server based time.

"now" is the client (browser) time.

With the "now - offset" we calculate the real server time, based on the initial time "offset", calculated on loading this page.

On loading page it was calculated the difference between server and browser as follow:
`offset =  browserTime - serverTime`

Every place you need "serverTime", you can calculate it by changing the formula above into:
`serverTime = browserTime - offset`

Of curse the primary change to save time only at server is the modification in php `function startRecorder` (see my step 8bd9dbc)
Without this modification, the stopwatch would display wrong count.

_23.04.2016: Text spelling updated_

Comment (HenryNe) [inline extensions/ki_timesheets/js/ts_func.js]: I must also say, that I not understand, why the previous variant was written so complicated. Before my modification the variable "startsec" contains a start sec relative (start with 0), if the buzzer was started from inside java. The same variable "startsec" contained an absolute time (based on 1970, I think), if the page was reloaded.

After my modification the "startsec" contains the absolute time of start, adjusted to the server base in all cases (start from buzzer, "start again" from time sheed, or start from reloading the page).
To get the counted time for buzzer, ticktac  must simple differ "startsec" from current time and respect the offset between server and local time.

File: js/main.js (hunk 2 above)
Comment (HenryNe) [inline js/main.js:482]: I must also say, that I not understand, why the previous variant was written so complicated.
Before my modification the variable "startsec" contains a start sec relative (start with 0), if the stopwatch was started from inside javascript. The same variable "startsec" contained an absolute time (based on 1970, I think), if the page was reloaded.

After my modification the "startsec" contains the absolute time of start, adjusted to the server base in all cases (start from buzzer, "start again" from time sheet, or start from reloading the page).
To get the counted time for stopwatch, ticktac must simple differ "startsec" from current time and respect the offset between server and local time.

_23.04.2016: updated spelling_

Comment (HenryNe) [inline js/main.js:482]: The formula in line 482 would be better readable in this form:

``` javascript
now_Local_based = Math.floor((new Date()).getTime()/1000);  /* current browser timestamp */
now_Server_based = now_Local_based - offset;       /* timestamp adjusted to server time */
start_Server_based = startsec;                     /* start time from server */
sek = now_Server_based - start_Server_based;       /* timediff from same base time */
```

File: extensions/ki_timesheets/js/ts_func.js (hunk 1 above)
Comment (simonschaufi) [inline extensions/ki_timesheets/js/ts_func.js]: declare new local variable with `var`

File: js/main.js (hunk 2 above)
Comment (simonschaufi) [inline js/main.js:482]: can we extract this calculation into a function? then you can use better internal variable names so that other people also understand what's going on.
I would expect something like this:

``` javascript
sek = getTimeDiff(...);
```

Comment (HenryNe) [inline js/main.js:482]: I want not to change to many things. It is a simple timediff calculation from global variables with different time bases.
I would prefer a comment for the calculation like this:

```
/* Timediff calculation:
   "sek" = ("now" as local time - "offset" to server) - "startsec" from server */
```

Comment (simonschaufi) [inline js/main.js:482]: I would prefer doing it. It makes the code much more readable and easier to test

Comment (kevinpapst) [inline js/main.js:482]: If you understand the code, please go ahead and add these comments (we have a lot of old code from various developer, that we have never touched before, so we are happy about any comment).

**Comment by HenryNe** [comment]
This pull request was splitted into two requests #684 and #685 now.
